plugin-server/tests/worker/ingestion/person-state.test.ts

```diff
@@ -2057,3 +2097,488 @@ describe('PersonState.update()', () => {
+                is_identified: true,
+            }),
+        ])
+
+        // verify Postgres distinct_ids
+        const distinctIds = await fetchDistinctIds()
+        expect(distinctIds).toEqual([
+            ['first', persons[0].id],
+            ['second', persons[0].id],
+        ])
+
+        // verify Postgres person_id overrides
+        const overrides = await fetchPersonIdOverrides()
+        expect(overrides).toEqual([[second.uuid, first.uuid]])
+
+        // verify running merge again doesn't change the state
+        await updatePersonStateFromEvent({
+            event: '$identify',
+            distinct_id: 'first',
+            properties: {
+                $anon_distinct_id: 'second',
+            },
+        })
+
+        const personsAgain = await fetchPostgresPersons()
+        const distinctIdsAgain = await fetchDistinctIds()
+        const overridesAgain = await fetchPersonIdOverrides()
+
+        expect(persons).toEqual(personsAgain)
+        expect(distinctIds).toEqual(distinctIdsAgain)
+        expect(overrides).toEqual(overridesAgain)
+    })
+
+    it('does not commit partial transactions on override conflicts', async () => {
+        await updatePersonStateFromEvent({
+            event: 'event',
+            distinct_id: 'first',
+        })
+
+        await updatePersonStateFromEvent({
+            event: 'event',
+            distinct_id: 'second',
+        })
+
+        const [first, second] = await fetchPostgresPersons()
+
+        const personsBeforeMergeAttempt = await fetchPostgresPersons()
+        const distinctIdsBeforeMergeAttempt = await fetchDistinctIds()
+        const overridesBeforeMergeAttempt = await fetchPersonIdOverrides()
+
+        const originalPostgresQuery = hub.db.postgresQuery.bind(hub.db)
+        const mockPostgresQuery = jest
+            .spyOn(hub.db, 'postgresQuery')
+            .mockImplementation(async (query: any, values: any[] | undefined, tag: string, ...args: any[]) => {
+                if (tag === 'transitivePersonOverrides') {
+                    throw new Error('Conflict')
+                }
+                return await originalPostgresQuery(query, values, tag, ...args)
+            })
+
+        await expect(
+            updatePersonStateFromEvent({
+                event: '$identify',
+                distinct_id: 'first',
+                properties: {
+                    $anon_distinct_id: 'second',
+                },
+            })
+        ).rejects.toThrow()
+
+        // verify Postgres persons
+        const personsAfterFailure = await fetchPostgresPersons()
+        expect(personsAfterFailure).toEqual(
+            personsBeforeMergeAttempt.map((person) =>
+                expect.objectContaining({
+                    id: person.id,
+                    uuid: person.uuid,
+                    properties: person.properties,
+                    created_at: person.created_at,
+                    // TODO: fix this. It seems that we update some person
```

Comment:
^

Reply:
Would you mind expanding on "^"?

This commit was also not added by me, but cherry picked from other PR (either yours or @hazzadous's).

So, if your "^" means "fix this what this comment says" then I'm going to need more context on that as I didn't write the comment myself.

Reply:
I don't know the context, but can look into it if needed, I'm not even sure it's true 🤷

Reply:
If you uncomment the lines is_identified and version, you'll end up with:

```
    +   Object {
    -     "is_identified": false,
    +     "is_identified": true,
...
    -     "version": "0",
    +     "version": "1",
        },
```

So if we for some reason fail to completely process the event, on retry we will see `is_identified === true` and thus we will not properly handle the retry as there is code that depends on the comparison. It's like that already, nothing to do with this PR so fine to leave.

The main thing to test is that running it again gets us to the right state which I think we do in this case. However, I _think_ there are cases where it will not be correct, e.g. if one of the persons was already identified, then the second becomes identified, then we can merge them afterwards on second run.

Reply:
aah I get it now, this is deterministic, so we can test this properly. We always mark the distinct_id person as `is_identified` even if merge failed. This helps make sure that anything that's a user_id won't be merged into anything else via `$identify` nor `$create_alias`.

So the test can be: 
```
distinct_id == 'first' person: is_identified: true, version: 1
distinct_id == 'second' person: is_identified: false, version: 0
```

Potentially easiest to change the code by adding a person property with the distinct_id values 'first' and 'second'